src/components/Table/TableBase.tsx

```diff
@@ -393,12 +401,14 @@ export class TableBase<T extends {}> extends React.Component<
 					: data.slice();
 		}
 		if (onCheck) {
-			onCheck(checkedItems);
+			onCheck(checkedItems, this.state.allChecked);
```

Comment:
I think that if we are in server side pagination and `this.toggleCheckedState()` returns `all`, then we shouldn't given checkedItems (in which case we also can skip computing them).

Reply:
Let me rephrase this, I think that for server side pagination, giving `checkedItems` & `all` at the same time should only happen when `checkedItems.length === pagination.totalItems`, since that's the only case that `checkedItems` will hold really `all` items.

Reply:
yep good point! If you agree I would just pass it as undefined wrapping the computation in a if statement